src/debug-panel.tsx

```
@@ -0,0 +1,194 @@
+  message: string;
+  timestamp: string;
+}
+
+const MAX_LOG_ENTRIES = 200;
+
+// Global log store shared across all instances
+const logEntries: LogEntry[] = [];
+let logListeners: Array<(entries: LogEntry[]) => void> = [];
+
+function notifyListeners() {
+  logListeners.forEach((l) => l([...logEntries]));
```

Biome (2.4.4)

[error] This callback passed to forEach() iterable method should not return a value.

(lint/suspicious/useIterableCallbackReturn)